binaryen.zig

```diff
@@ -35,7 +35,7 @@ pub const Module = opaque {
     pub fn emitBinary(self: *Module, source_map_url: ?[*:0]const u8) EmitBinaryResult {
         const result = byn.BinaryenModuleAllocateAndWrite(self.c(), source_map_url);
         return .{
-            .binary = @ptrCast([*]u8, result.binary)[0..result.binaryBytes],
+            .binary = @as([*]u8, @ptrCast(result.binary))[0..result.binaryBytes],
```

Comment:
Cleaner to split this into a separate `binary_ptr` variable with a type annotation, rather than using `@as`